src/checker.rs

```diff
@@ -987,8 +987,18 @@ impl<'a, F: Function> Checker<'a, F> {
         let mut queue = Vec::new();
         let mut queue_set = FxHashSet::default();
 
-        queue.push(self.f.entry_block());
-        queue_set.insert(self.f.entry_block());
+        // Put every block in the queue to start with, to ensure
+        // everything is visited even if the initial state remains
+        // `Top` after preds update it.
+        for block in 0..self.f.num_blocks() {
```

Comment:
How would you feel about adding the blocks in reverse order to begin with, rather than having to reverse the array afterward?

```suggestion
        for block in (0..self.f.num_blocks()).rev() {
```

Reply:
Done, thanks!